apps/sequencer/src/providers/provider.rs

```diff
@@ -369,19 +370,20 @@ impl RpcProvider {
 
     pub async fn get_latest_values(
         &self,
-        feed_ids: &[u32],
+        feed_ids: &[FeedId],
     ) -> Result<Vec<Result<PublishedFeedUpdate, PublishedFeedUpdateError>>, eyre::Error> {
         let multicall = self.get_contract_address(MULTICALL_CONTRACT_NAME)?;
         let data_feed = self.get_contract_address(PRICE_FEED_CONTRACT_NAME)?;
         let contract = MulticallInstance::new(multicall, self.provider.clone());
         let calldata: Vec<Multicall::Call> = feed_ids
             .iter()
             .map(|feed_id| {
+                let feed_id = *feed_id as u32;
```

Comment:
What if it is bigger then u32? Isn't this the reason we are moving from u32?

Reply:
I think this functionality for reading from the contract is only relevant to the old version of the contracts. In the old version of the contracts feed id-s are 4 bytes. This is also why when we call legacy serialize we cast to u32 as well.

Reply:
@HristoStaykov said he will refactor the Sequencer test to work with ADFS and will test for u128::MAX in another PR.